examples/classification/benchmark.py

```diff
@@ -0,0 +1,384 @@
+                        stub to the model for DeepSparse or ONNX runtime
+                        benchmarks.
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -e {deepsparse,onnxruntime}, --engine {deepsparse,onnxruntime}
+                        Inference engine backend to run benchmark on. Choices
+                        are 'deepsparse', and 'onnxruntime'. Default is
+                        'deepsparse'
+  --data-path DATA_PATH
+                        path to SparseZoo stub for data sample to benchmark
+                        with, or to directory of .npz files with named sample
+                        inputs to the given model, inputs should be for a
+                        single sample with no batch dimension included.
+                        Defaults to benchmark on random image samples with
+                        channel size 3.
+  --image-shape IMAGE_SHAPE [IMAGE_SHAPE ...]
+                        Image shape to benchmark with, must be two integers.
+                        If None model is used to infer image shape
+  -b BATCH_SIZE, --batch-size BATCH_SIZE
+                        The batch size to run the benchmark for
+  -c NUM_CORES, --num-cores NUM_CORES
+                        The number of physical cores to run the benchmark on,
+                        defaults to None where it uses all physical cores
+                        available on the system. For DeepSparse benchmarks,
+                        this value is the number of cores per socket
+  -s NUM_SOCKETS, --num-sockets NUM_SOCKETS
+                        For DeepSparse benchmarks only. The number of physical
+                        cores to run the benchmark on. Defaults to None where
+                        is uses all sockets available on the system
+  -i NUM_ITERATIONS, --num-iterations NUM_ITERATIONS
+                        The number of iterations the benchmark will be run for
+  -w NUM_WARMUP_ITERATIONS, --num-warmup-iterations NUM_WARMUP_ITERATIONS
+                        The number of warmup iterations that will be executed
+                        before the actual benchmarking
+#####
+Example
+python benchmark.py \
+    "zoo:cv/classification/resnet_v1-50/pytorch/sparseml/imagenet/pruned-moderate"
+"""
+import argparse
+import time
+from collections import namedtuple
+from tempfile import NamedTemporaryFile
+from typing import Any, List, Optional, Tuple, Union
+
+import numpy
+import onnx
+import onnxruntime
+from tqdm import tqdm
+
+from deepsparse import compile_model
+from deepsparse.benchmark import BenchmarkResults
+from examples.classification.helper import _BatchLoader, _load_random_data, load_data
+from sparseml.onnx.utils import (
+    get_tensor_dim_shape,
+    override_model_batch_size,
+    set_tensor_dim_shape,
+)
+from sparsezoo import Zoo
+
+
+DEEPSPARSE_ENGINE = "deepsparse"
+ORT_ENGINE = "onnxruntime"
+
+_Timer = namedtuple("_Timer", ("start", "end"))
```

Comment:
nit: add typing for named tuple parameters ([example](https://github.com/neuralmagic/sparseml/blob/e7e11d000135e38c47cacfd60912a1f2fdbfb5ca/src/sparseml/pytorch/models/registry.py#L39))

Reply:
Thanks for the example, check latest push! 🌼